processor/src/main/java/org/mapstruct/ap/internal/model/source/MappingOptions.java

```diff
@@ -74,7 +80,14 @@ public static MappingOptions empty() {
      * @return MappingOptions with only regular mappings
      */
     public static MappingOptions forMappingsOnly( Map<String, List<Mapping>> mappings ) {
```

Comment:
Maybe we need to pass the `BeanMapping`. I think that we only need to explicitly ignore when we have multiple source parameters.